config/initializers/freeagent/invoice.rb

```diff
@@ -0,0 +1,7 @@
+module FreeAgent
+  class Invoice < Resource
+    def self.filter_by_date(from, to)
+      FreeAgent::Invoice.filter(nested_invoice_items: 'true').reject { |invoice| invoice.due_on < from || invoice.due_on > to }
```

Comment:
`due_on` is not the right field to check here... we need the creation date.

Also, this is definitely not an initializer, it's a monkey patch... so we should move it to `lib` and explicitely name it `freeagent_invoice_monkeypatch.rb` or `freeagent_invoice_filter_by_date.rb`, fmpov.

Also, to make it a bit less dirty, we shoud make a module out of it and then include it into `FreeAgent::Invoice` rather than reopening the class. See here for an exemple of how to do it: https://github.com/Locale/localeapp/blob/master/lib/localeapp/rails/force_exception_handler_in_translation_helper.rb